test/functional/ui/incsubstitution_spec.lua

```diff
@@ -0,0 +1,379 @@
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      ^these ARE nothing                       |
+      with colorful text (ARE)                |
+      with colorful text (ARE)                |
+      1001 substitutions on 1001 lines        |
+    ]])
+
+    feed('i')
+    feed('example of insertion')
+
+    screen:expect([[
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      these ARE nothing                       |
+      with colorful text (ARE)                |
+      example of insertion^these ARE nothing   |
+      with colorful text (ARE)                |
+      with colorful text (ARE)                |
+      {6:-- INSERT --}                            |
+   ]])
+
+   feed('<esc>')
+   execute('set incsubstitute=2')
+   execute('1,1000s/ARE/nut/g')
+   feed(':%s/ARE/to')
+
+   screen:expect([[
+     with colorful text (to)                 |
+     these to nothing                        |
+     with colorful text (to)                 |
+     these to nothing                        |
+     with colorful text (to)                 |
+     these to nothing                        |
+     with colorful text (to)                 |
+     example of insertionthese to nothing    |
+     with colorful text (to)                 |
+     with colorful text (to)                 |
+     {11:[No Name] [+]                           }|
+      [1001]with colorful text ({12:to})          |
+      [1002]these {12:to} nothing                 |
+      [1003]with colorful text ({12:to})          |
+      [1004]these {12:to} nothing                 |
+      [1005]with colorful text ({12:to})          |
+      [1006]these {12:to} nothing                 |
+      [1007]with colorful text ({12:to})          |
+     {10:[inc_sub]                               }|
+     :%s/ARE/to^                              |
+   ]])
+  end)
+
+end)
```

Comment:
Also missing splitview tests with more then one window (in particular, after `vsplit | split` and after `split | vsplit`): where splitview window will appear?

Also is it affected by &splitbelow, &splitright (unlikely), &equalalways, &eadirection options? This should also be tested.